frontend/test/metabase/scenarios/dashboard/reproductions/21830-slow-loading-card-viz-options-error.cy.spec.js

```diff
@@ -0,0 +1,40 @@
+import {
+  restore,
+  editDashboard,
+  showDashboardCardActions,
+} from "__support__/e2e/cypress";
+
+describe.skip("issue 21830", () => {
+  beforeEach(() => {
+    restore("postgres-12");
+    cy.signInAsAdmin();
+  });
+
+  it("slow loading card visualization options click shouldn't lead to error (metabase#21830)", () => {
+    cy.intercept(
+      {
+        method: "POST",
+        url: "/api/dashboard/*/dashcard/*/card/*/query",
+        middleware: true,
+      },
+      req => {
+        req.on("response", res => {
+          // Throttle the response to 500 Kbps to simulate a mobile 3G connection
+          res.setThrottle(500);
+        });
+      },
+    ).as("dashcardQuery");
+
+    cy.visit("/dashboard/1");
+
+    // It's crucial that we try to click on this icon BEFORE we wait for the `dashcardQuery` response!
+    editDashboard();
+    showDashboardCardActions();
+
+    cy.icon("palette").click({ force: true });
```

Comment:
This will fail because actions would be hidden until the card is loaded

Reply:
@ranquild I cannot know the implementation in advance. This is the way to currently reproduce the issue. In general, I'd merge it and then when you apply a fix, you can modify the repro.

But since you've given me a glimpse of how do you want to approach this, I can modify it now.
Which icons will be hidden? All (including `X`) or visualization options, add series button and click behavior?